### PR TITLE
fix: don't download blocks older than anchor slot in SyncBlocks

### PR DESCRIPTION
**Pull request by MegaRedHand**
This PR fixes a race condition we had. If during start-up we received the anchor block's parent, we marked that one as pending, and thus also marked our current anchor block as pending, causing all block processing to stop.

### CODE REVIEW

File: lib/lambda_ethereum_consensus/fork_choice/simple_tree.ex [hunk 1]
```diff
@@ -3,7 +3,7 @@ defmodule LambdaEthereumConsensus.ForkChoice.Simple.Tree do
 
   defmodule Node do
     @moduledoc false
-    defstruct [:parent_id, :id, :children_ids]
+    defstruct [:parent_id, :children_ids]
```
Comment (MegaRedHand) [inline lib/lambda_ethereum_consensus/fork_choice/simple_tree.ex:6]: This attribute was unused

**Comment by MegaRedHand** [comment]
I removed the check. The problem was caused by `sync_blocks` downloading blocks older than the anchor 😅

File: lib/lambda_ethereum_consensus/beacon/sync_blocks.ex [hunk 2]
```diff
@@ -24,7 +24,7 @@ defmodule LambdaEthereumConsensus.Beacon.SyncBlocks do
     # Initial sleep for faster app start
     Process.sleep(1000)
     checkpoint = BeaconChain.get_finalized_checkpoint()
-    initial_slot = Misc.compute_start_slot_at_epoch(checkpoint.epoch)
+    initial_slot = Misc.compute_start_slot_at_epoch(checkpoint.epoch) + 1
```
Comment (mpaulucci) [inline lib/lambda_ethereum_consensus/beacon/sync_blocks.ex:27]: 😁

**Review by mpaulucci — COMMENTED** [review]
nice changes! just a small comment regarding block state

File: lib/lambda_ethereum_consensus/beacon/beacon_chain.ex [hunk 3]
```diff
@@ -92,6 +90,11 @@ defmodule LambdaEthereumConsensus.Beacon.BeaconChain do
   def init({anchor_state = %BeaconState{}, time}) do
     schedule_next_tick()
 
+    anchor_checkpoint = %Checkpoint{
+      root: anchor_state.latest_block_header |> Ssz.hash_tree_root!(),
```
[the remark this answers is not in the document]
Comment (mpaulucci) [reply lib/lambda_ethereum_consensus/beacon/beacon_chain.ex:94]: we probably want to move that function to a utility

[the remark this answers is not in the document]
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/beacon/beacon_chain.ex]: Nice catch!

**Review by mpaulucci — APPROVED** [review]
🎉

**Pull request merged**